bindings/gumjs/gumquickmemory.c

```diff
@@ -931,21 +932,63 @@ GUMJS_DEFINE_FUNCTION (gumjs_memory_scan)
   GumMemoryScanContext sc;
   gpointer address;
   gsize size;
-  const gchar * match_str;
+  JSValue pattern_val;
+  JSValue callbacks;
+
+  if (args->count < 4)
+    return _gum_quick_throw_literal (ctx, "missing argument");
+
+  if (!_gum_quick_native_pointer_get (ctx, args->elements[0], core,
+      &address))
+    return JS_EXCEPTION;
 
-  if (!_gum_quick_args_parse (args, "pZsF{onMatch,onError?,onComplete}",
-      &address, &size, &match_str, &sc.on_match, &sc.on_error, &sc.on_complete))
+  if (!_gum_quick_size_get (ctx, args->elements[1], core, &size))
     return JS_EXCEPTION;
+
+  pattern_val = args->elements[2];
+
+  if (JS_IsString (pattern_val)) {
+    const gchar * match_str;
+
+    if (!_gum_quick_string_get (ctx, args->elements[2], &match_str))
+      return JS_EXCEPTION;
+
+    sc.pattern = gum_match_pattern_new_from_string (match_str);
+    JS_FreeCString (ctx, match_str);
+  } else if (JS_IsObject (pattern_val)) {
+    sc.pattern = JS_GetOpaque(pattern_val, core->match_pattern_class);
+  } else {
+    return _gum_quick_throw_literal (ctx,
+        "expected either a pattern string or a MatchPattern object");
+  }
+
+  if (sc.pattern == NULL)
+    return _gum_quick_throw_literal (ctx, "invalid match pattern");
+
+  callbacks = args->elements[3];
+
+  if (!JS_IsObject (callbacks)) {
+    return _gum_quick_throw_literal (ctx,
+        "expected an object containing callbacks");
+  }
+
+  sc.on_match = JS_GetPropertyStr (ctx, callbacks, "onMatch");
+  sc.on_complete = JS_GetPropertyStr (ctx, callbacks, "onComplete");
+
+  if (!JS_IsFunction (ctx, sc.on_match)) {
+    return _gum_quick_throw_literal (ctx, "expected a callback value");
+  }
```

Comment:
Braces should be omitted for single-line statements. (Unless the line is really long/code is hard to read.)